basics_of_logging_in_sst-core/Example00/ExampleComponent.cc

```diff
@@ -0,0 +1,97 @@
+// Created for SST-Core Version (9.1.0)
+//
+#include "ExampleComponent.h"
+#include <iostream>
+
+using namespace Example00;
+
+// Component constructor.
+//
+ExampleComponent::ExampleComponent(SST::ComponentId_t id, SST::Params &params) :
+    SST::Component(id),
+    componentId_(id),
```

Comment:
In general, you should avoid using or storing the component's id.  If it were possible to make the id unavailable to the end component, we would have done that.  The id should just be passed to the parent constructor.  The id is a value that is generally meaningless to the end user because it is assigned opaquely by the core with no guarantees other than it is unique on each rank.  In anything other than a trivial simulation, the number will not allow the user to know which component it refers to.  And, while this is a trivial example, it is better not to teach this as something that is acceptable.

Better in this example would likely be to use the getName() function that exists in the BaseComponent API.  This will return the name given to the component in the python file and will be meaningful to the end use.

Reply:
The above comment applies through all of the files that store the component id.